ocaml/idl/datamodel_repository.ml

```diff
@@ -70,6 +71,9 @@ let t =
     ~contents:
       [ uid       _repository ~lifecycle;
         namespace ~name:"name" ~contents:(names ~writer_roles:_R_POOL_OP None RW) ();
+        field   ~qualifier:DynamicRO ~lifecycle
+          ~ty:(Set (Ref _pool)) "enabled_on"
+          "The pools where the repository is enabled on";
```

Comment:
There is always just one pool in the DB, so this could just be a single `ref`. And then is this field actually useful?

Reply:
But if you make it a single ref, you get a one-to-many relation, where you can only set the "one" side, which is this field. And that will make the pool setters go wrong. Personally, I'd just remove this field.

Reply:
This `enabled_on` was added after I got error like:
```
gen_api_main ocaml/xapi-client/client.ml (exit 2)
(cd _build/default/ocaml/xapi-client && ../idl/ocaml_backend/gen_api_main.exe -filterinternal true -filter closed -mode client -output client.ml)
Fatal error: exception (Failure "Set(Ref _) field is not in relations table: pool.repositories")
```
And I had to add relation between them. Thus a "enabled_on" field in Repository. (But just as you pointed out, single ref of "enabled_on" makes the pool setters go wrong. And again I had to change type of "enabled_on" to set as well).

Back to the origin, it looks like the set "repositories" of pool does require a relation? May I get a pointer on how to avoid the error above without relation?

Reply:
Is it happening because you added the relation: `(_pool, "repositories"), (_repository, "enabled_on");`?

I'm also wondering why the API calls `add_repository`, `remove_repository`, `set_repositories` were added to datamodel_pool.ml and not datamodel_repository.ml

Reply:
I think that adding `~ignore_foreign_key:true` to your `set ref` field definition will avoid this. It tells the code generator not to expect a relation.

Reply:
Hi Rob, with your help, the relation was dropped by adding `~ignore_foreign_key:true` in `pool.repositories` now. Thanks.

Reply:
Hi Ben, I think the `add_repository`, `remove_repository` and `set_repositories` are pool level methods. So I added them into `datamodel_pool.ml`.